### PR TITLE
feat: add `utils/none-in-by`

### LINKED ISSUE #824: [RFC]: Add `@stdlib/utils/none-in-by`
### Description

This RFC proposes adding a utility to test whether every "own" and inherited property of a provided object does not satisfy a predicate function.

This is the `for-in` equivalent of [`@stdlib/utils/none-by`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/none-by).

Package: `@stdlib/utils/none-in-by`
Alias: `noneInBy`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by AgPriyanshu18**
Check whether every property of object does not satisfy predicate function

Fixes #824

Resolves #824 .

## Description

> What is the purpose of this pull request?

This pull request:

-   a add utils/none-in-by
-   b Check whether every property of object does not satisfy predicate function
-   c

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #824 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/utils/none-in-by/benchmark/julia/benchmark.jl [hunk 1]
```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env julia
+#
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
Comment (Jaysukh-409) [inline lib/node_modules/@stdlib/utils/none-in-by/benchmark/julia/benchmark.jl]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

**Review by Jaysukh-409 — COMMENTED** [review]
Change Copyright years wherever it is necessary.

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+	function isPositive(value) {
+		return (value > 0);
+	}
+
+	bool = noneInBy(obj, isPositive);
+
+	t.strictEqual(bool, false, 'returns false');
+	t.end();
+});
```
Comment (Jaysukh-409) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js:134]: Needs to add test for providing an execution context.

File: lib/node_modules/@stdlib/utils/none-in-by/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+var obj = {
+    'a': 12,
+    'b': 22,
+    'c': 25
+};
+
+var bool = noneInBy( obj, isUnderage );
+// returns false
+```
+
```
Comment (Jaysukh-409) [inline lib/node_modules/@stdlib/utils/none-in-by/README.md:78]: Add example of the function execution context and what predicate function receives as an argument.

**Comment by AgPriyanshu18** [comment]
@Jaysukh-409 I have made all suggested changes.

File: lib/node_modules/@stdlib/utils/none-in-by/README.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# noneInBy
+
+> Tests whether every property of an object fails a test implemented by a predicate function.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/utils/none-in-by/README.md]:
```suggestion
> Test whether every property of an object fails a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/none-in-by/README.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+-   If provided an empty object, the function returns `true`.
+
+```javascript
+function truthy() {
+    return true;
+}
+var bool = noneInBy( {}, truthy );
+// returns true
+```
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/utils/none-in-by/README.md]:
```suggestion
    ```javascript
    function truthy() {
        return true;
    }
    var bool = noneInBy( {}, truthy );
    // returns true
    ```
```

File: lib/node_modules/@stdlib/utils/none-in-by/README.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/utils/none-in-by/README.md:164]: mIssing ``The function does **not** skip `undefined` elements.``

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
This PR needs a few changes, I have requested changes and ask you to iterate over entire package comparing against `stdlib/utils/none-by`. Thank you!

**Comment by AgPriyanshu18** [comment]
@Pranavchiku I have completed the suggested changes and have reviewed the whole package.

File: lib/node_modules/@stdlib/utils/none-in-by/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,42 @@
+
+{{alias}}( object, predicate[, thisArg ] )
+    Tests whether every property in a collection fail a test implemented by a
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/docs/repl.txt]:
```suggestion
    Tests whether every property in an object fails a test implemented by a
```

File: lib/node_modules/@stdlib/utils/none-in-by/docs/types/test.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+	return ( v > 0 );
+};
+
+// TESTS //
+
+var obj = {
+	'a':1,
+	'b':2,
+	'c':4
+};
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/docs/types/test.ts]:
```suggestion
const obj = {
	'a': 1,
	'b': 2,
	'c': 4
};
```

File: lib/node_modules/@stdlib/utils/none-in-by/lib/index.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Test whether every property of a object fail a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/lib/index.js]:
```suggestion
* Test whether every property of an object fails a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/none-in-by/lib/index.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+* @module @stdlib/utils/none-in-by
+*
+* @example
+* var noneInBy = require( '@stdlib/utils/none-in-by' );
+*
+* function isPositive( v ) {
+*     return ( v > 0 );
+* }
+*
+* var obj = { 'a' : -1, 'b' : -2, 'c' : -3, 'd' : -4};
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/lib/index.js]:
```suggestion
* var obj = { 'a' : -1, 'b' : -2, 'c' : -3, 'd' : -4 };
```

File: lib/node_modules/@stdlib/utils/none-in-by/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+// MAIN //
+
+/**
+* Tests whether every property of a provided object does not satisfy a predicate function.
+*
+* @param {Object} obj - input object
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @throws {TypeError} first argument must be a object
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/lib/main.js]:
```suggestion
* @throws {TypeError} first argument must be an object
```

File: lib/node_modules/@stdlib/utils/none-in-by/lib/main.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+* var obj = { 'a' : -1, 'b' : -2, 'c' : -3, 'd' : -4};
+*
+* var bool = noneInBy( obj, isPositive );
+* // returns true
+*/
+function noneInBy( obj, predicate, thisArg ) {
+	var key;
+
+	if ( !isObject( obj ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a object. Value: `%s`.', obj ) );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/lib/main.js]:
```suggestion
		throw new TypeError( format( 'invalid argument. First argument must be an object. Value: `%s`.', obj ) );
```

File: lib/node_modules/@stdlib/utils/none-in-by/package.json [hunk 13]
```diff
@@ -0,0 +1,71 @@
+{
+    "name": "@stdlib/utils/none-in-by",
+    "version": "0.0.0",
+    "description": "Test whether every property in a collection fail a test implemented by a predicate function.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/package.json]:
```suggestion
    "description": "Test whether every property in an object fails a test implemented by a predicate function.",
```

File: lib/node_modules/@stdlib/utils/none-in-by/package.json [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+      "util",
+      "test",
+      "predicate",
+      "none",
+      "all",
+      "every",
+      "array.every",
+      "iterate",
+      "collection",
+      "array-like",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/package.json]:
```suggestion
      "every",
      "iterate",
      "object",
```

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+		t.fail('should not be invoked');
+	}
+
+	bool = noneInBy(obj, foo);
+
+	t.strictEqual(bool, true, 'returns true');
+	t.end();
+});
+
+tape('the function returns `true` if every property fail a test ', function test(t) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js]:
```suggestion
tape('the function returns `true` if every property fails a test ', function test(t) {
```

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof noneInBy, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape('the function throws an error if not provided an object', function test(t) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js]:
```suggestion
tape( 'the function throws an error if not provided an object', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			noneInBy( value, value );
+		};
+	}
+});
+
+tape('if provided an empty object, the function returns `true`', function test(t) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js]:
```suggestion
tape( 'if provided an empty object, the function returns `true`', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+		return (value > 0);
+	}
+
+	bool = noneInBy(obj, isPositive);
+
+	t.strictEqual(bool, true, 'returns true');
+	t.end();
+});
+
+tape('the function returns `false` if one or more properties pass a test ', function test(t) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js]:
```suggestion
tape( 'the function returns `false` if one or more properties pass a test ', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/none-in-by/test/test.js [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+		'2': 3.0
+	};
+
+	bool = noneInBy(obj, sum, ctx);
+
+	t.strictEqual(bool, true, 'returns true');
+	t.strictEqual(ctx.sum / ctx.count, 2.0, 'expected result');
+
+	t.end();
+});
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/none-in-by/test/test.js]: Should update spacing around parentheses per our code style:

```suggestion
tape( 'the function supports providing an execution context', function test( t ) {
	var bool;
	var ctx;
	var obj;

	function sum( value ) {
		/* eslint-disable no-invalid-this */
		if ( value < 0 ) {
			return false;
		}
		this.sum += value;
		this.count += 1;
		return false;
	}

	ctx = {
		'sum': 0,
		'count': 0
	};
	obj = {
		'0': 1.0,
		'1': 2.0,
		'2': 3.0
	};

	bool = noneInBy( obj, sum, ctx );

	t.strictEqual( bool, true, 'returns true' );
	t.strictEqual( ctx.sum / ctx.count, 2.0, 'expected result' );

	t.end();
});
```

**Review by Planeshifter — APPROVED** [review]
Looks good; just made a few code style and documentation fixes, but otherwise this PR should be ready to land once CI clears.

**Pull request merged**